### PR TITLE
Channel test1

### PR DESCRIPTION
**Pull request by zpatel-dev**
Addedjustchat project in prototype1 folder

this communication uses Django channels which are basically WebSockets.

In order to run this chat project, you need to install docker. for opening a port and other stuff.
refer to this link for more details about Redis in docker: https://channels.readthedocs.io/en/latest/tutorial/part_2.html

After installing docker run this: `docker run -p 6379:6379 -d redis:5`
Also need to install: python3 -m pip install channels_redis

changed checkers_rules_engine.py to python3 (because my Django is running on python3)

merge chat app functionality with our main project. create mysite/routing.py, checkers/routing.py, also modified views and URL to accommodate this.

### CODE REVIEW

File: mysite/checkers/views.py [hunk 1]
```diff
@@ -89,3 +93,15 @@ class game(View):
     def get(self, request):
         if request.method == 'GET':
             return HttpResponse(render(request,'checkers/game.html'))
+
```
Comment (brucker3) [inline mysite/checkers/views.py:96]: new views should be in there own class

File: mysite/checkers/consumers.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# chat/consumers.py
```
Comment (brucker3) [inline mysite/checkers/consumers.py:1]: can you add some commets to this code what is this new class for and what does it do? also new classes should have unit tet

Comment (brucker3) [reply mysite/checkers/consumers.py:1]: test*

File: mysite/mysite/views.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from django.http import HttpResponse
```
Comment (brucker3) [inline mysite/mysite/views.py]: why was this file added?

File: mysite/templates/homepage.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
Comment (brucker3) [inline mysite/templates/homepage.html]: why do we need another home page?

File: mysite/templates/login.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (brucker3) [inline mysite/templates/login.html]: we already have a log in page

File: prototype_1/justchat/chat/consumers.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# chat/consumers.py
```
Comment (brucker3) [inline prototype_1/justchat/chat/consumers.py]: why are we adding to prototype_1? the code should all run thru the djago app?

**Comment by brucker3** [comment]
I added a lot of comments in areas of the pr I have questions about. this pr is really big so its going to take a while to go thru. I also did not see where or why we needed to add docker? the app runs fine on local host with Django.

File: mysite/checkers/views.py (hunk 1 above)
Comment (zpatel-dev) [reply mysite/checkers/views.py:96]: Room is just part of game like create session.

File: mysite/templates/login.html (hunk 5 above)
Comment (zpatel-dev) [reply mysite/templates/login.html]: removed

**Comment by zpatel-dev** [comment]
Sorry, my bad! many unnecessary files were there just removed it.  Justchat project was for reference. (removed)

File: mysite/checkers/consumers.py (hunk 2 above)
Comment (zpatel-dev) [reply mysite/checkers/consumers.py:1]: I still need to learn unit test and add it.

Comment (brucker3) [reply mysite/checkers/consumers.py:1]: ok no worries we can write another story for that and then go over unit test later

File: mysite/static/checkers_script.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
+/*----------- Game State Data ----------*/
```
Comment (brucker3) [inline mysite/static/checkers_script.js:1]: what are theses static/ files for?

**Comment by brucker3** [comment]
looks a lot better with the unneeded files removed. Ill look thru it again later tonight

**Review by brucker3 — APPROVED** [review]
